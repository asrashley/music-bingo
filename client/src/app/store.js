import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { connectRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { reducer as formReducer } from 'redux-form';

import cardsReducer from '../cards/cardsSlice';
import gamesReducer from '../games/gamesSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import adminReducer from '../admin/adminSlice';
import usersMiddleware from '../user/userMiddleware';

export const history = createBrowserHistory();

export const store = configureStore({
  middleware: [...getDefaultMiddleware(), usersMiddleware, logger],
  reducer: {
    admin: adminReducer,
    cards: cardsReducer,
    form: formReducer,
    games: gamesReducer,
    tickets: ticketsReducer,
    router: connectRouter(history),
    user: userReducer,
  },
});
