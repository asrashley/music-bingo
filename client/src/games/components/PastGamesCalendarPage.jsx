import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { PastGamesCalendar, PastGamesCalendarPropType } from './PastGamesCalendar';
import { PastGamesButtons } from './PastGamesButtons';

import { fetchUserIfNeeded } from '../../user/userSlice';
import {
    fetchGamesIfNeeded, invalidateGames,
} from '../gamesSlice';

import { getPastGamesCalendar } from '../gamesSelectors';
import { getUser } from '../../user/userSelectors';

import { UserPropType } from '../../user/types/User';

class PastGamesCalendarPageComponent extends React.Component {
    static propTypes = {
        dispatch: PropTypes.func.isRequired,
        user: UserPropType.isRequired,
        pastGamesCalendar: PastGamesCalendarPropType.isRequired,
    };

    componentDidMount() {
        const { dispatch } = this.props;
        dispatch(fetchUserIfNeeded());
        dispatch(fetchGamesIfNeeded());
    }

    componentDidUpdate(prevProps, prevState) {
        const { user, dispatch } = this.props;
        if (user.pk > 0 && user.pk !== prevProps.user.pk) {
            dispatch(fetchGamesIfNeeded());
        }
    }

    onReload = () => {
        const { dispatch } = this.props;
        dispatch(invalidateGames());
        dispatch(fetchGamesIfNeeded());
    };

    render() {
        const { user } = this.props;
        const { themes, months } = this.props.pastGamesCalendar;

        return (
            <div id="games-page" className={user.loggedIn ? '' : 'modal-open'}  >
                <PastGamesButtons page="calendar" />
                <PastGamesCalendar themes={themes} months={months} />
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        user: getUser(state, ownProps),
        pastGamesCalendar: getPastGamesCalendar(state, ownProps),
    };
};

export const PastGamesCalendarPage = connect(mapStateToProps)(PastGamesCalendarPageComponent);
