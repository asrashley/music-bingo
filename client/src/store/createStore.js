import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { connectRouter } from 'connected-react-router';

import adminReducer from '../admin/adminSlice';
import directoriesReducer from '../directories/directoriesSlice';
import gamesReducer from '../games/gamesSlice';
import messagesReducer from '../messages/messagesSlice';
import settingsReducer from '../settings/settingsSlice';
import systemReducer from '../system/systemSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import usersMiddleware from '../user/userMiddleware';
import messagesMiddleware from '../messages/messagesMiddleware';

import { history } from './history';

export function createStore(preloadedState, logging) {
  const middleware = [...getDefaultMiddleware(), usersMiddleware, messagesMiddleware];
  if (logging === undefined) {
    logging = (process.env.NODE_ENV === 'development');
  }
  if (logging) {
    middleware.push(logger);
  }
  return configureStore({
    middleware,
    reducer: {
      admin: adminReducer,
      directories: directoriesReducer,
      games: gamesReducer,
      messages: messagesReducer,
      settings: settingsReducer,
      system: systemReducer,
      tickets: ticketsReducer,
      router: connectRouter(history),
      user: userReducer,
    },
    preloadedState
  });
}
