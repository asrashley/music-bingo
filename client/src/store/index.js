export { createStore } from './createStore';
export { initialState } from './initialState';
export { history } from './history';
