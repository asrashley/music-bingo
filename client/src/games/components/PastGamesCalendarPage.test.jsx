import React from 'react';
import { screen } from '@testing-library/react';
import log from 'loglevel';

import { fetchMock, renderWithProviders } from '../../../tests';
import { PastGamesCalendarPage } from './PastGamesCalendarPage';
import { MockBingoServer, normalUser } from '../../../tests/MockServer';
import { createStore, initialState } from '../../store';


describe('PastGamesCalendarPage component', () => {
    afterEach(() => {
        fetchMock.mockReset();
        log.resetLevel();
    });

    it('to render a calendar previous games', async () => {
        //log.setLevel('debug');
        const mockServer = new MockBingoServer(fetchMock, { currentUser: normalUser });
        const store = createStore({
            ...initialState,
            user: mockServer.getUserState(normalUser),
        });
        const { asFragment } = renderWithProviders(<PastGamesCalendarPage />, { store });
        await screen.findAllByText("Rock & Power Ballads", { exact: false });
        await fetchMock.flush(true);
        expect(asFragment()).toMatchSnapshot();
    });
});