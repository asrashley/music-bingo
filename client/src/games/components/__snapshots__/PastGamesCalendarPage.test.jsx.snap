// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastGamesCalendarPage component to render a calendar previous games 1`] = `
<DocumentFragment>
  <div
    class=""
    id="games-page"
  >
    <div
      class="past-games-buttons"
    >
      <div
        class="row mb-4 page-calendar"
      >
        <div
          class="col text-center"
        >
          <a
            class="btn btn-lg btn-light border border-dark-subtle popularity-button"
            href="/history"
          >
            <h4>
              Theme Popularity
            </h4>
          </a>
        </div>
        <div
          class="col text-center"
        >
          <a
            class="btn btn-lg btn-light border border-dark-subtle usage-button"
            href="/history/themes"
          >
            <h4>
              Theme Last Usage
            </h4>
          </a>
        </div>
        <div
          class="col text-center"
        >
          <a
            class="btn btn-lg btn-light border border-dark-subtle calendar-button"
            href="/history/calendar"
          >
            <h4>
              Games Calendar
            </h4>
          </a>
        </div>
        <div
          class="col text-center"
        >
          <a
            class="btn btn-lg btn-light border border-dark-subtle all-button"
            href="/history/games"
          >
            <h4>
              All Games
            </h4>
          </a>
        </div>
      </div>
    </div>
    <table
      class="past-games-calendar table table-striped"
    >
      <thead>
        <tr
          class="year-row"
        >
          <th
            class="theme-title"
            rowspan="2"
          >
            Theme
          </th>
          <th
            class="year"
            colspan="2"
          >
            2018
          </th>
          <th
            class="year"
            colspan="1"
          >
            2020
          </th>
          <th
            class="theme-title"
            rowspan="2"
          >
            Theme
          </th>
        </tr>
        <tr
          class="month-row"
        >
          <th
            class="month"
          >
            04
          </th>
          <th
            class="month"
          >
            05
          </th>
          <th
            class="month"
          >
            06
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/motown"
            >
              Motown
            </a>
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/motown"
            >
              Motown
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/pot-luck"
            >
              Pot Luck
            </a>
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/pot-luck"
            >
              Pot Luck
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/pride"
            >
              Pride
            </a>
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/pride"
            >
              Pride
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/rock"
            >
              Rock
            </a>
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/rock"
            >
              Rock
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/rock-power-ballads"
            >
              Rock & Power Ballads
            </a>
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/rock-power-ballads"
            >
              Rock & Power Ballads
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/sunshine"
            >
              sunshine
            </a>
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/sunshine"
            >
              sunshine
            </a>
          </td>
        </tr>
        <tr>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/various-artists"
            >
              Various Artists
            </a>
          </td>
          <td
            class="calendar-cell active"
          >
            1
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="calendar-cell inactive"
          >
            0
          </td>
          <td
            class="theme-title"
          >
            <a
              href="/history/themes/various-artists"
            >
              Various Artists
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
